### PR TITLE
feat: enable load callbacks

### PR DESCRIPTION
**Pull request by pimlie**
Resolves: #240 

This pr adds the possibility for adding callbacks which are called when eg the script has been loaded. This makes it possible to chain scripts which are depending on each other.

## Client side implementation
When the element is created the onload attribute is just directly set

## Server side implementation
- Elements need to have a `${tagIDKeyName}` set (`vmid` / `hid`)
- When a tag has a callback, an onload attribute is added `onload="this.__vm_l=1"`
- On hydration, VueMeta will check for the `${ssrAttribute}` attribute
- When this exists, the load callbacks are added
- The corresponding callback for an element is resolved using the `tagIDKeyName`
- On existing elements an load event listener is directly added
- When the DOM is not fully loaded, an event listener is added onReadyStateChange which adds callback to available elements
- Once the load callback is triggered for a tag, the onload attribute will be removed from the element

VueMeta uses the following three props directly on the element for hydration/ load callback control:
- _`__vm_cb`_ whether the load callback has been called
- _`__vm_l`_ set by onload attribute, whether the element was loaded
- _`__vm_ev`_ whether the event listener was added or not

## `skip` attribute
I have added a `skip` attribute for tag definitions in metaInfo so its easier to do reactive chaining (see async-callback example)

## Potential caveats
- ~~Callbacks on tags which are loaded before the main app will probably not trigger~~ not anymore due to onload attribute which is always added

## ToDo
- [x] add tests

### CODE REVIEW

**Comment by galvez** [comment]
daaaamn

You've been busy :D

File: examples/async-callback/app.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          vmid: 'vegetables',
+          src: '/user-2.js',
+          async: true,
+          callback: this.updateCounter
+        },
+        {
+          vmid: 'meat',
+          src: '/user-1.js',
+          async: true,
+          callback: el => this.loadCallback(el)
```
[the remark this answers is not in the document]
Comment (pimlie) [reply examples/async-callback/app.js]: Actually this verbosity was on purpose as I wanted to show that the target element is passed as an argument. People never read anything and without this I am sure they will probably overlook the loadCallback definition and wont see the element being used :)

[the remark this answers is not in the document]
Comment (pimlie) [reply examples/async-callback/app.js]: I changed the fingerprint of loadCallback so it expects the vmid as argument, would that be ok for you?

**Comment by pimlie** [comment]
Due to issues in IE9, I've opted for a more basic approach. Its quite ugly if you ask me, but adding the onload attribute on ssr makes sure that we know the element has been loaded also when the loading takes place _before_ we can add a load event listener.

I've also removed the MutationObserver and just try to add callbacks on every document.readyState change.

File: src/client/load.js [hunk 2]
```diff
@@ -93,10 +70,46 @@ export function applyCallbacks (dataAttributeName, matchElement) {
     }
 
     for (const element of elements) {
-      element.addEventListener('load', () => {
-        element.removeAttribute(`data-${dataAttributeName}`)
+      /* __vm_cb: whether the load callback has been called
+       * __vm_l: set by onload attribute, whether the element was loaded
+       * __vm_ev: whether the event listener was added or not
+       */
+      if (element.__vm_cb) {
```
Comment (pimlie) [inline src/client/load.js]: I've could have made this an object like `__vm = {}` instead but as `__vm_l` needs to be set from the onload attribute and using an object there is too complicated,  this approach of separate variables seemed ok to me.

**Pull request merged**